subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/ScriptPluginsScope.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.use.resolve.internal;
+
+import org.gradle.api.internal.initialization.ClassLoaderScope;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.api.internal.project.ProjectRegistry;
+
+public class ScriptPluginsScope {
+
+    public static ClassLoaderScope from(ProjectRegistry<ProjectInternal> projectRegistry) {
```

Comment:
We shouldn't use a static method here that makes assumptions about the arrangement of classloader scopes constructed elsewhere. Instead, should use a service that can explicitly answer the question about the class loader scopes, eg "give me the classloader scope for script plugins", "give me the classloader scope to use for buildSrc", etc.

This way the logic for the arrangement of the scopes is in one place (that service) and is explicit, rather than scattered around the place.

Reply:
> We shouldn't use a static method here that makes assumptions about the arrangement of classloader scopes constructed elsewhere.

The function is just a way to avoid duplicating the logic temporarily as this will go away once script plugins get more sophisticated in subsequent stories.